modules/tensor_mechanics/src/actions/TensorMechanicsAction.C

```diff
@@ -4,18 +4,33 @@
 /*          All contents are licensed under LGPL V2.1           */
 /*             See LICENSE for full restrictions                */
 /****************************************************************/
-#include "TensorMechanicsAction.h"
-#include "Factory.h"
-#include "FEProblem.h"
 #include "Conversion.h"
 #include "FEProblem.h"
 #include "Factory.h"
 #include "MooseMesh.h"
 #include "MooseObjectAction.h"
+#include "TensorMechanicsAction.h"
 
 #include "libmesh/string_to_enum.h"
 #include <algorithm>
 
+const std::map<std::string, std::string> TensorMechanicsAction::_ranktwoaux_table = {
+    {"strain", "total_strain"},
+    {"stress", "stress"},
+    {"elastic_strain", "elastic_strain"},
+    {"plastic_strain", "plastic_strain"},
+    {"creep_strain", "creep_strain"}};
+const std::vector<char> TensorMechanicsAction::_component_table = {'x', 'y', 'z'};
+const std::map<std::string, std::string> TensorMechanicsAction::_ranktwoscalaraux_table = {
+    {"vonmises", "VonMisesStress"},
+    {"hydrostatic", "Hydrostatic"},
+    {"stress_max", "MaxPrincipal"},
+    {"stress_mid", "MidPrincipal"},
+    {"stress_min", "MinPrincipal"},
+    {"firstinv", "FirstInvariant"},
```

Comment:
I think we need to have `stress` in all of these names to clarify what they are.  Otherwise, you don't know what tensor `firstinv` is the first invariant of.  I think I'd prefer to name these as `stress_vonmises`, `stress_hydrostatic`, `stress_firstinv`, etc.  It sounds better to put `_stress` at the end of these, but putting it at the beginning is more consistent with the component forms.  I could go either way on whether `stress` is at the front or end.